modules/cloudsql-instance/variables.tf

```diff
@@ -143,6 +143,26 @@ variable "labels" {
   default     = null
 }
 
+variable "maintenance_config" {
+  description = "Set maintenance window configuration and maintenance deny period (up to 90 days). Date format: 'yyyy-mm-dd'."
+  type = object({
+    maintenance_window = optional(object({
+      day          = number
+      hour         = number
+      update_track = optional(string, "stable")
```

Comment:
tqo things here
- if this is a "string as an enum", meaning it can have multiple fixed values, we need a validation condition that ensures only possible values (or `null`) are passed in
- if the provider uses `stable` as default, we leave the module's default to `null` and let the provider infer it